internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -124,18 +125,29 @@ func expectedProxyContainers(infra *ir.ProxyInfra, deploymentConfig *egcfgv1a1.K
 	}
 
 	var bootstrapConfigurations string
+
+	// Get the default Bootstrap
+	defaultBootstrapConfigurations, err := bootstrap.GetRenderedBootstrapConfig(proxyMetrics)
+	if err != nil {
+		return nil, err
+	}
+
 	// Get Bootstrap from EnvoyProxy API if set by the user
 	// The config should have been validated already
 	if infra.Config != nil &&
 		infra.Config.Spec.Bootstrap != nil {
-		bootstrapConfigurations = *infra.Config.Spec.Bootstrap
-	} else {
-		var err error
-		// Use the default Bootstrap
-		bootstrapConfigurations, err = bootstrap.GetRenderedBootstrapConfig(proxyMetrics)
-		if err != nil {
-			return nil, err
+		userBootstrapConfigurations := infra.Config.Spec.Bootstrap.Value
```

Comment:
seeing this style used in many places, can we add a `util` in `internal/xds/bootstrap` to improve reuse ?